spec/presenters/tree_builder_ops_rbac_spec.rb

```diff
@@ -83,4 +83,17 @@ def x_get_tree_custom_kids_for_and_expect_objects(type_of_model, expected_object
       end
     end
   end
+
+  describe "#x_get_tree_tenant_kids" do
```

Comment:
better however you are testing a code which I suggested to removed - you can still reuse the test for the "count" part.

and tests are better but it is still not reflect DB with 2 region, you can image it like two duplicated DB with different IDs, so one object in one region exists like duplicated object in other region.
You tests are mixing it together(like one tenant has children from other region) which is probably testing  issue but it is not same as in real. SO I am suggesting to do it like it could be in real.

So one thing what you need to have is Tenant.root in other region.


```ruby
     let!(:child_tenant) { FactoryGirl.create(:tenant, :parent => Tenant.root_tenant) }
     let(:inactive_region) { FactoryGirl.create(:miq_region) }
     let(:tenant_root_in_inactive_region) do
      root_in_inactive_region = Tenant.root_tenant.dup
      root_in_inactive_region.id = Tenant.id_in_region(Tenant.count + 1_000_000, inactive_region.region) # you can create a method (DRY)
     root_in_inactive_region.save
     root_in_inactive_region
    end


     let!(:child_tenant_out_of_region) { FactoryGirl.create(:tenant, :parent => tenant_root_in_inactive_region, :id => Tenant.id_in_region(Tenant.count + 1_000_000, inactive_region.region)) }
```

Reply:
I guess you didn't try that out or I'm missing something :) It's impossible to create second root `Tenant` in specs. It fails at `Validation failed: Tenant: Parent required`. I'll use a tenant that's in different region but has a parent from active region. Spec checks that count is correct not their relationship.

Is that ok with you? :)

Reply:
I see, you are right. It about that validator use current region. I think we can skip validation with 
` root_in_inactive_region.save!(:validate => false)` in this case and you can leave parent as nil.

thanks!